{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "\n",
    "def read_logs(uid):\n",
    "    history = []\n",
    "    with open(file_name) as reader:\n",
    "        line = reader.readline().split('->')[-1]\n",
    "        val = json.loads(line)\n",
    "        if uid == val.get('name'):\n",
    "            history.append(val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['tasks',\n",
       " 'mlog.py',\n",
       " 'app-thread.py',\n",
       " 'app.log',\n",
       " 'main.py',\n",
       " '.ipynb_checkpoints',\n",
       " '__pycache__',\n",
       " 'Dockerfile',\n",
       " 'app.py',\n",
       " 'app.ini',\n",
       " '.idea',\n",
       " 'test.py',\n",
       " 'training_model.py',\n",
       " 'Untitled.ipynb']"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.listdir()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# with open('readme.txt') as f:\n",
    "#     lines = f.readlines()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-03-10 10:48:22,937 ->Started\n",
      "\n",
      "2021-03-10 10:48:22,944 ->{\"name\": \"anomaly_48fd8459f38d3692ba1f\", \"status\": \"in progress\", \"operation\": \"Data loading\"}\n",
      "\n",
      "2021-03-10 10:48:49,874 ->Exception on /status/anomaly_4234bbffb4d4c1b8dc11 [GET]\n",
      "\n",
      "Traceback (most recent call last):\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 2447, in wsgi_app\n",
      "\n",
      "    response = self.full_dispatch_request()\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 1952, in full_dispatch_request\n",
      "\n",
      "    rv = self.handle_user_exception(e)\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 1821, in handle_user_exception\n",
      "\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/_compat.py\", line 39, in reraise\n",
      "\n",
      "    raise value\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 1950, in full_dispatch_request\n",
      "\n",
      "    rv = self.dispatch_request()\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 1936, in dispatch_request\n",
      "\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "\n",
      "  File \"app-thread.py\", line 43, in spool_status\n",
      "\n",
      "    history = custom_log.read_logs(handle)\n",
      "\n",
      "  File \"./mlog.py\", line 34, in read_logs\n",
      "\n",
      "    val = json.loads(line)\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/json/__init__.py\", line 348, in loads\n",
      "\n",
      "    return _default_decoder.decode(s)\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/json/decoder.py\", line 337, in decode\n",
      "\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/json/decoder.py\", line 355, in raw_decode\n",
      "\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "\n",
      "2021-03-10 10:49:02,424 ->{\"name\": \"anomaly_48fd8459f38d3692ba1f\", \"status\": \"in progress\", \"operation\": \"Data loaded\"}\n",
      "\n",
      "2021-03-10 10:49:02,424 ->{\"name\": \"anomaly_48fd8459f38d3692ba1f\", \"status\": \"in progress\", \"operation\": \"Model loading\"}\n",
      "\n",
      "2021-03-10 10:49:03,167 ->From /home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.\n",
      "\n",
      "Instructions for updating:\n",
      "\n",
      "Colocations handled automatically by placer.\n",
      "\n",
      "2021-03-10 10:49:03,347 ->From /home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/tensorflow/python/ops/math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.\n",
      "\n",
      "Instructions for updating:\n",
      "\n",
      "Use tf.cast instead.\n",
      "\n",
      "2021-03-10 10:49:03,764 ->{\"name\": \"anomaly_48fd8459f38d3692ba1f\", \"status\": \"in progress\", \"operation\": \"Model loaded\"}\n",
      "\n",
      "2021-03-10 10:49:03,764 ->{\"name\": \"anomaly_48fd8459f38d3692ba1f\", \"status\": \"in progress\", \"operation\": \"Model training\"}\n",
      "\n",
      "2021-03-10 10:51:52,123 ->{\"name\": \"anomaly_48fd8459f38d3692ba1f\", \"status\": \"in progress\", \"operation\": \"Model trained\"}\n",
      "\n",
      "2021-03-10 10:51:52,291 ->{\"name\": \"anomaly_48fd8459f38d3692ba1f\", \"status\": \"completed\", \"operation\": \"Model saved\", \"location\": \"../../../../../Data_set/packages/anomaly_48fd8459f38d3692ba1f\"}\n",
      "\n",
      "2021-03-10 11:32:43,124 ->Started\n",
      "\n",
      "2021-03-10 11:33:00,239 ->{\"name\": \"anomaly_4e9e94095d6e6a517dfa\", \"status\": \"in progress\", \"operation\": \"Data loading\"}\n",
      "\n",
      "2021-03-10 11:33:18,038 ->Exception on /status/anomaly_4e9e94095d6e6a517dfa [GET]\n",
      "\n",
      "Traceback (most recent call last):\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 2447, in wsgi_app\n",
      "\n",
      "    response = self.full_dispatch_request()\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 1952, in full_dispatch_request\n",
      "\n",
      "    rv = self.handle_user_exception(e)\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 1821, in handle_user_exception\n",
      "\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/_compat.py\", line 39, in reraise\n",
      "\n",
      "    raise value\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 1950, in full_dispatch_request\n",
      "\n",
      "    rv = self.dispatch_request()\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 1936, in dispatch_request\n",
      "\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "\n",
      "  File \"app-thread.py\", line 43, in spool_status\n",
      "\n",
      "    history = custom_log.read_logs(handle)\n",
      "\n",
      "  File \"./mlog.py\", line 34, in read_logs\n",
      "\n",
      "    val = json.loads(line)\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/json/__init__.py\", line 348, in loads\n",
      "\n",
      "    return _default_decoder.decode(s)\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/json/decoder.py\", line 337, in decode\n",
      "\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/json/decoder.py\", line 355, in raw_decode\n",
      "\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "\n",
      "2021-03-10 11:33:39,874 ->{\"name\": \"anomaly_4e9e94095d6e6a517dfa\", \"status\": \"in progress\", \"operation\": \"Data loaded\"}\n",
      "\n",
      "2021-03-10 11:33:39,874 ->{\"name\": \"anomaly_4e9e94095d6e6a517dfa\", \"status\": \"in progress\", \"operation\": \"Model loading\"}\n",
      "\n",
      "2021-03-10 11:33:40,652 ->From /home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.\n",
      "\n",
      "Instructions for updating:\n",
      "\n",
      "Colocations handled automatically by placer.\n",
      "\n",
      "2021-03-10 11:33:40,835 ->From /home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/tensorflow/python/ops/math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.\n",
      "\n",
      "Instructions for updating:\n",
      "\n",
      "Use tf.cast instead.\n",
      "\n",
      "2021-03-10 11:33:41,262 ->{\"name\": \"anomaly_4e9e94095d6e6a517dfa\", \"status\": \"in progress\", \"operation\": \"Model loaded\"}\n",
      "\n",
      "2021-03-10 11:33:41,262 ->{\"name\": \"anomaly_4e9e94095d6e6a517dfa\", \"status\": \"in progress\", \"operation\": \"Model training\"}\n",
      "\n",
      "2021-03-10 11:34:34,045 ->Exception on /status/anomaly_4e9e94095d6e6a517dfa [GET]\n",
      "\n",
      "Traceback (most recent call last):\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 2447, in wsgi_app\n",
      "\n",
      "    response = self.full_dispatch_request()\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 1952, in full_dispatch_request\n",
      "\n",
      "    rv = self.handle_user_exception(e)\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 1821, in handle_user_exception\n",
      "\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/_compat.py\", line 39, in reraise\n",
      "\n",
      "    raise value\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 1950, in full_dispatch_request\n",
      "\n",
      "    rv = self.dispatch_request()\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/site-packages/flask/app.py\", line 1936, in dispatch_request\n",
      "\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "\n",
      "  File \"app-thread.py\", line 43, in spool_status\n",
      "\n",
      "    history = custom_log.read_logs(handle)\n",
      "\n",
      "  File \"./mlog.py\", line 34, in read_logs\n",
      "\n",
      "    val = json.loads(line)\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/json/__init__.py\", line 348, in loads\n",
      "\n",
      "    return _default_decoder.decode(s)\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/json/decoder.py\", line 337, in decode\n",
      "\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "\n",
      "  File \"/home/ahmad/anaconda3/envs/modelupdate/lib/python3.7/json/decoder.py\", line 355, in raw_decode\n",
      "\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "\n",
      "2021-03-10 11:36:15,825 ->{\"name\": \"anomaly_4e9e94095d6e6a517dfa\", \"status\": \"in progress\", \"operation\": \"Model trained\"}\n",
      "\n",
      "2021-03-10 11:36:15,884 ->{\"name\": \"anomaly_4e9e94095d6e6a517dfa\", \"status\": \"completed\", \"operation\": \"Model saved\", \"location\": \"../../../../../Data_set/packages/anomaly_4e9e94095d6e6a517dfa\"}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "file_name = 'app.log'\n",
    "# read_logs('anomaly_4e9e94095d6e6a517dfa')\n",
    "history = []\n",
    "with open('app.log') as reader:\n",
    "    lines = reader.readlines()\n",
    "    for line in lines:\n",
    "        print(line)\n",
    "#         val = \n",
    "#         val = json.loads(line.split('->')[-1])\n",
    "        \n",
    "#     pri\n",
    "#     if uid == val.get('name'):\n",
    "#         history.append(val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hello\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "\n",
    "import sqlalchemy\n",
    "\n",
    "# sqlalchemy.__version__\n",
    "# from sqlalchemy import create_engine\n",
    "# engine = create_engine('sqlite:///college.db', echo=True)\n",
    "# # dialect[+driver]://user:password@host/dbname\n",
    "# engine.connect()\n",
    "\n",
    "from sqlalchemy import (create_engine, MetaData,\n",
    "                        Table, Column, String,\n",
    "                        Binary, ARRAY, BIGINT)\n",
    "\n",
    "\n",
    "sql_alchemy_conn_str = \"postgresql://postgres:somePassword@localhost:5432/postgres\"\n",
    "\n",
    "engine = create_engine(sql_alchemy_conn_str, echo=False)\n",
    "\n",
    "meta = MetaData()\n",
    "\n",
    "ml_model = Table(\n",
    "    'spool_table', meta,\n",
    "    Column('uid', String, primary_key=True),\n",
    "    Column('status', String),\n",
    "    Column('operation', String),\n",
    "    Column(\"location\", String),\n",
    "\n",
    "    # Column('class', String),\n",
    "    # Column('model', Binary),\n",
    "    # Column('owner', String),\n",
    "    # Column('description', String),\n",
    "    # Column('timestamp', BIGINT),\n",
    ")\n",
    "# custom_log.write_log(uid, 'completed', operation='Model saved', location=temp_storage)\n",
    "\n",
    "\n",
    "def return_model_table_name():\n",
    "    meta.create_all(engine)\n",
    "    return 'spool_table'\n",
    "\n",
    "\n",
    "def wirte_enery():\n",
    "    pass\n",
    "\n",
    "def read_enery():\n",
    "    pass\n",
    "\n",
    "def foo():\n",
    "    pass\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    print('hello')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import sqlalchemy\n",
    "\n",
    "meta.create_all(engine)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MetaData(bind=None)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
